torch/_subclasses/fake_tensor.py

```diff
@@ -1393,6 +1405,16 @@ def _cache_key(
             # where it wasn't seen on a previous instance of the same op.
             self.shape_env.settings if self.shape_env else None,
         ]
+
+        if isinstance(func, torch._ops.HigherOrderOperator):
+            # For invoke_subgraph, ignore the subgraph arg. We rely on
```

Comment:
why do we need to do this here ? cant the op impl / op caching handle this?

Reply:
There's a subgraph here. We're trying to avoid needing to hash the subgraph everytime it is seen. So that's what the identifier is for.

Taking suggestions for how to make this better

Reply:
If the subgraph is repeated, can we have FakeTensorCache cache the hashing of it ?

Reply:
I like that better, what say you @anijain2305 ?

As a part of this, you could define what it means for a "subgraph to be cacheable" to be what the current "invoke subgraph fake tensor cache validator function" is and get rid of the "fake tensor cache validator registration" thing.